sdks/python/apache_beam/runners/dataflow/internal/apiclient.py

```diff
@@ -588,11 +588,25 @@ def _stage_resources(self, pipeline, options):
               'Found duplicated artifact: %s (%s)',
               type_payload.path,
               type_payload.sha256)
+          staged_name = hashs[type_payload.sha256]
           dep.role_payload = beam_runner_api_pb2.ArtifactStagingToRolePayload(
-              staged_name=hashs[type_payload.sha256]).SerializeToString()
+              staged_name=staged_name).SerializeToString()
         else:
-          resources.append((type_payload.path, role_payload.staged_name))
-          hashs[type_payload.sha256] = role_payload.staged_name
+          staged_name = role_payload.staged_name
+          resources.append((type_payload.path, staged_name))
+          hashs[type_payload.sha256] = staged_name
+
+        if google_cloud_options.staging_location.startswith('gs://'):
```

Comment:
Is this check needed ? I think Dataflow staging location always has to be a GCS path.

Reply:
Yes, but still we can't blindly update the artifact type to URL artifact (for example, Dataflow local runner uses a local directory as a staging location).

Reply:
Might be better to use:

if FileSystems.get_scheme(google_cloud_options.staging_location) == GcsFileSystem.scheme():
  ...

Reply:
Done.